### PR TITLE
Merge 9.2

### PR DESCRIPTION
**Pull request by tgauth**
# PR Summary
Pull in upstream changes

## PR Context
https://www.openssh.com/txt/release-9.2

### CODE REVIEW

File: scp.c [hunk 1]
```diff
@@ -770,6 +775,16 @@ main(int argc, char **argv)
 					     optarg + 7, strerror(errno));
 				}
 				sftp_copy_buflen = (size_t)llv;
+#ifdef WINDOWS
+				size_t max_sftp_buf_len = 204800;
```
Comment (vthiebaut10) [inline scp.c]: Maybe have a constant for that.

File: scp.c [hunk 2]
```diff
@@ -184,7 +184,12 @@ pid_t do_cmd_pid = -1;
 pid_t do_cmd_pid2 = -1;
 
 /* SFTP copy parameters */
+#ifdef WINDOWS
+// match DEFAULT_COPY_BUFLEN from sftp-client.c unless overriden below with -X input
+size_t sftp_copy_buflen = 32768;
```
Comment (vthiebaut10) [inline scp.c]: Have this number in a constant too

File: regress/connection-timeout.sh [hunk 3]
```diff
@@ -1,6 +1,12 @@
 #	$OpenBSD: connection-timeout.sh,v 1.2 2023/01/17 10:15:10 djm Exp $
 #	Placed in the Public Domain.
 
+if [ "$os" == "windows" ]; then
```
Comment (vthiebaut10) [inline regress/connection-timeout.sh:4]: We might want to add a pester test for this before we release

File: regress/channel-timeout.sh [hunk 4]
```diff
@@ -9,7 +9,7 @@ rm -f $OBJ/sshd_proxy.orig
 cp $OBJ/sshd_proxy $OBJ/sshd_proxy.orig
 
 verbose "no timeout"
-${SSH} -F $OBJ/ssh_proxy somehost "sleep 5 ; exit 23"
+${SSH} -vvv -F $OBJ/ssh_proxy somehost "sleep 5 ; exit 23"
```
Comment (vthiebaut10) [inline regress/channel-timeout.sh]: This -vvv was just for debugging, right? Probably want to remove this,

File: regress/connection-timeout.sh (hunk 3 above)
Comment (vthiebaut10) [inline regress/connection-timeout.sh:4]: We might want to add a pester test to this before we release

Comment (tgauth) [reply regress/connection-timeout.sh:4]: per offline discussion, leaving that as backlog item

**Review by vthiebaut10 — APPROVED** [review]
We are good to go :)

**Review by PaulHigin — APPROVED** [review]

**Pull request merged**